common4j/src/main/com/microsoft/identity/common/java/controllers/ExceptionAdapter.java

```diff
@@ -222,11 +222,25 @@ private static boolean shouldBeConvertedToUiRequiredException(final String oAuth
      * @param errorResponse
      * @return ServiceException, UiRequiredException
      */
-    public static ServiceException getExceptionFromTokenErrorResponse(@NonNull final TokenErrorResponse errorResponse) {
+    static ServiceException getExceptionFromTokenErrorResponse(@NonNull final TokenErrorResponse errorResponse) {
+        return getExceptionFromTokenErrorResponse(errorResponse, true);
+    }
+
+    /**
+     * Get an exception object from the given oAuth values.
+     *
+     * @param errorResponse - TokenErrorResponse containing error information
+     * @param allowConvertToUiRequiredException - if true, will convert to UiRequiredException if applicable
+     * @return ServiceException, UiRequiredException
+     */
+    static ServiceException getExceptionFromTokenErrorResponse(
+            @NonNull final TokenErrorResponse errorResponse,
+            final boolean allowConvertToUiRequiredException
+    ) {
 
         final ServiceException outErr;
 
-        if (shouldBeConvertedToUiRequiredException(errorResponse.getError())) {
+        if (allowConvertToUiRequiredException && shouldBeConvertedToUiRequiredException(errorResponse.getError())) {
```

Comment:
Where will the value of the parameter be set to false for Silent token requests?

Reply:
It's false by default. It will be set in brokered flow if the request is for resource account. That will be separate PR.